core/wiki/macros/toc.tid

```diff
@@ -2,182 +2,214 @@ title: $:/core/macros/toc
 tags: $:/tags/Macro
 
 \define toc-caption()
-<$set name="tv-wikilinks" value="no">
-  <$transclude field="caption">
-    <$view field="title"/>
-  </$transclude>
-</$set>
+\whitespace trim
+<$set name="tv-wikilinks" value=no>
+  <$list filter="[all[current]has[caption]]" 
+      emptyMessage="""<$transclude field=title><$view field=title/></$transclude>""">
+     <$transclude field=caption>
+       <$view field="title"/>
+     </$transclude>
+  </$list>
+</$set> 
 \end
 
-\define toc-body(tag,sort:"",itemClassFilter,exclude,path)
+\define getItemClass() [all[current]]-[<tv-history-list>get[focused-tiddler]]
```

Comment:
It would be clearer to add a single space before the `-`: 

```
\define getItemClass() [all[current]] -[<tv-history-list>get[focused-tiddler]]
```

Reply:
Again, I don't really understand what's going on here. Why are we touching the `$:/HistoryList` tiddler at all? Using the TOC is nothing to do with navigation (until the point a user clicks on a link).

Reply:
The history tiddler is updated, whenever a link is clicked. So it contains the actual status, which can be used to calculate the `toc-item-selected` status